frontend/src/lib/logic/featureFlagLogic.ts

```diff
@@ -21,17 +22,24 @@ function notifyFlagIfNeeded(flag: string, flagState: boolean): void {
     }
 }
 
+function getPersistedFeatureFlags(): FeatureFlagsSet {
+    const persistedFeatureFlags = getAppContext()?.persist_feature_flags || []
+    return Object.fromEntries(persistedFeatureFlags.map((f) => [f, true]))
+}
+
 function spyOnFeatureFlags(featureFlags: FeatureFlagsSet): FeatureFlagsSet {
+    const combinedFlags = { ...featureFlags, ...getPersistedFeatureFlags() }
+
     if (typeof window.Proxy !== 'undefined') {
         return new Proxy(
             {},
             {
                 get(_, flag) {
                     if (flag === 'toJSON') {
-                        return JSON.stringify(featureFlags)
+                        return () => combinedFlags
```

Comment:
Context into why this doesn't return the object JSON-stringified?

Reply:
I made a mistake before. This should return a function that returns a JSON-serializable object
https://dev.to/jankapunkt/how-stringify-proxy-to-json-10oe